tiledb/sm/cpp_api/array.h

```diff
@@ -341,6 +345,7 @@ class Array {
     ctx.handle_error(tiledb_array_get_schema(c_ctx, carray, &array_schema));
     schema_ = ArraySchema(ctx, array_schema);
 
+    dont_close_ = dont_close;
```

Comment:
@stavrospapadopoulos @Shelnutt2 I think we decided against this in the subarray api refactor PR?

Reply:
The functionality was not needed there, here it is needed to avoid unnecessary destruction/recreation of arrays in use in the unit tests, and I expect ultimately within user code as well.  While it can be removed, it creates unnecessary extra work for user's of the class with 'non-owned' arrays and will require reworking of some unit test(s).

It -is- actually the appropriate functionality for the class with a 'non-owned' array, but as it has already been released, it could cause problems to simply change the default functionality to not close a 'non-owned' array.